### PR TITLE
task 9372: send transaction screen - handle event

### PR DESCRIPTION
**Pull request by vanpd-0859**
### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/9372
### Goals :soccer:

- Validate all textfields
- Click paste button to paste clipboard to that textfield
- Click QR button to go to QR controller to scan QR code
- Click Max to fill out the maximum available amount of that asset
- Click USD to switch to Amount USD
- Click Next to go to Confirm Send Transaction Screen

### Implementation Details :construction:

https://gfycat.com/ScientificSleepyFairyfly

### CODE REVIEW

**Comment by vanpd-0859** [comment]
ready

File: CryptoWallet/Repositories/EthereumInteraction.swift [hunk 1]
```diff
@@ -125,7 +125,24 @@ enum EthereumInteraction {
...
+
+    static func getEstimatedFee() -> (estimatedFee: Double, gasPrice: Double)? {
+        let web3 = Web3.InfuraMainnetWeb3()
+        do {
+            let gasPrice = try web3.eth.getGasPrice()
+            let fee = gasPrice * 21_000
+            guard let stringFee = Web3.Utils.formatToEthereumUnits(fee,
+                                                                   toUnits: .eth,
+                                                                   decimals: 18),
+                let doubleFee = Double(stringFee) else {
```
Comment (tgdong2296) [inline CryptoWallet/Repositories/EthereumInteraction.swift]:
```suggestion
                let fee = Double(stringFee) else {
```
để tên biến như kia đọc dễ hiểu nhầm thành nhân 2 giá lắm :v

File: CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift [hunk 2]
```diff
@@ -72,7 +72,7 @@ final class AssetHeader: UIView {
     func setAssetInfo(_ assetInfo: AssetInfo) {
         self.assetInfo = assetInfo
         assetAmountLabel.do {
-            $0.text = String(format: "%.4f", assetInfo.amount) + String(assetInfo.symbol)
+            $0.text = String(format: "%.4f", assetInfo.amount) + " " + String(assetInfo.symbol)
```
Comment (tgdong2296) [inline CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift]: cái format string cho vào extension của String

File: CryptoWallet/Repositories/EthereumInteraction.swift [hunk 3]
```diff
@@ -125,7 +125,24 @@ enum EthereumInteraction {
...
+        } catch {
+            return nil
+        }
+    }
+
+    static func getEstimatedFee() -> (estimatedFee: Double, gasPrice: Double)? {
+        let web3 = Web3.InfuraMainnetWeb3()
+        do {
+            let gasPrice = try web3.eth.getGasPrice()
+            let fee = gasPrice * 21_000
```
Comment (tienpx3010) [inline CryptoWallet/Repositories/EthereumInteraction.swift]: Bỏ vào Constants

File: CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift (hunk 2 above)
Comment (tienpx3010) [reply CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift]:
```suggestion
            $0.text = String(format: "%.4f %@", assetInfo.amount, assetInfo.symbol)
```

File: CryptoWallet/Utils/ValueType.swift [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+//  ValueType.swift
+//  CryptoWallet
+//
+//  Created by Phan Dinh Van on 4/16/19.
+//  Copyright © 2019 Phan Dinh Van. All rights reserved.
+//
+
+import Foundation
+
+enum ValueType {
```
Comment (tienpx3010) [inline CryptoWallet/Utils/ValueType.swift]: Bỏ vào Model

Comment (vanpd-0859) [reply CryptoWallet/Utils/ValueType.swift]: enum cũng bỏ vào model hả a?

**Comment by vanpd-0859** [comment]
ready

**Review by tienpx3010 — APPROVED** [review]

**Review by tgdong2296 — APPROVED** [review]

**Pull request merged**